File: src/main/java/org/railwaystations/rsapi/core/model/Photo.java
package org.railwaystations.rsapi.core.model;

import jakarta.validation.constraints.NotNull;
import lombok.Builder;
import lombok.Value;

import java.time.Instant;

@Value
@Builder
public class Photo implements Comparable<Photo> {

    long id;
    Station.Key stationKey;
    boolean primary;
    String urlPath;
    User photographer;
    Instant createdAt;
    License license;
    boolean outdated;

    @Override
    public int compareTo(@NotNull Photo o) {
        if (primary && !o.isPrimary()) {
            return -1;
        }
        if (!primary && o.isPrimary()) {
            return 1;
        }
        return Long.compare(id, o.getId());
    }

}

File: src/main/kotlin/org/railwaystations/rsapi/core/model/Photo.kt
package org.railwaystations.rsapi.core.model

import java.time.Instant

data class Photo(
    var id: Long = 0,
    var stationKey: Station.Key,
    var primary: Boolean = false,
    var urlPath: String,
    var photographer: User,
    var createdAt: Instant,
    var license: License,
    var outdated: Boolean = false,
) : Comparable<Photo> {

    override fun compareTo(other: Photo): Int {
        if (primary && other.primary) {
            return -1
        }
        if (!primary && other.primary) {
            return 1
        }
        return id.compareTo(other.id)
    }
}
